### PR TITLE
#449 - Remove front end generic form

### PR DESCRIPTION
**Pull request by jozefbakus**
Implements: #449 
- Created 3 components: Spark, Hyperdrive, Shell

### CODE REVIEW

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  jobChanges: Subject<WorkflowEntryModel> = new Subject<WorkflowEntryModel>();
+  jobChangesSubscription: Subscription;
+
+  constructor() {
+    // do nothing
+  }
+
+  ngOnInit(): void {
+    this.jobChangesSubscription = this.jobChanges.subscribe((sensorChange) => {
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts]: Please rename `sensorChange`

File: ui/src/app/components/workflows/workflow-form/jobs/job/job.component.ts [hunk 2]
```diff
@@ -75,15 +82,10 @@ export class JobComponent implements OnInit, OnDestroy {
   }
 
   getSelectedJob() {
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/job/job.component.ts]: I see this method name was like this before, but it's really confusing. Shouldn't it be `getSelectedJobTemplateId`?

File: ui/src/app/components/workflows/workflow-form/jobs/jobs.component.ts [hunk 3]
```diff
@@ -25,6 +25,7 @@ import {
   WorkflowRemoveJob,
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/jobs/jobs.component.ts:25]: In `getJobName`
```
const nameOption = jobData.find((value) => value.property === this.workflowFormParts.staticJobPart.property);
```
Here, couldn't you replace `this.workflowFormParts.staticJobPart.property` with a constant `"name"`? Similarly how you did it in `job.component.ts` with `jobTemplateId`

Otherwise, you can't get rid of `workflowFormParts`

File: ui/src/app/services/workflow/workflow.service.ts [hunk 4]
```diff
@@ -165,14 +165,14 @@ export class WorkflowService {
...
-  getWorkflowDynamicFormParts(): Observable<DynamicFormParts> {
+  getWorkflowDynamicFormParts(): Observable<DynamicFormPartsResponse> {
     return this.getJobTemplates().pipe(
       mergeMap((jobTemplates) => {
         const sensorParts = WorkflowService.getSensorDynamicFormParts();
         const jobParts = jobTemplates.map((jobTemplate) => {
           return WorkflowService.getJobDynamicFormPart(jobTemplate);
         });
-        return of(DynamicFormPartsFactory.create(sensorParts, jobParts));
+        return of(DynamicFormPartsResponseFactory.create(DynamicFormPartsFactory.create(sensorParts, jobParts), jobTemplates));
```
Comment (kevinwallimann) [inline ui/src/app/services/workflow/workflow.service.ts:175]: Do we still need `sensorParts` and `jobParts`? I thought all of that information should be now be hard-coded in the respective components

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts (hunk 1 above)
Comment (kevinwallimann) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts]: Same in `shell.component.ts` and `spark.component.ts`

File: ui/src/app/components/workflows/workflow-form/dynamic-parts/string-sequence-part/string-sequence-part.component.ts [hunk 5]
```diff
@@ -50,8 +50,11 @@ export class StringSequencePartComponent implements OnInit {
       this.partValidation?.maxLength ?? Number.MAX_SAFE_INTEGER,
       this.partValidation?.minLength ?? 1,
     );
+    this.setDefaultValue();
+  }
 
-    if (!this.value) this.modelChanged(this.partValidationSafe.isRequired ? [''] : []);
+  ngOnChanges(changes: SimpleChanges): void {
```
Comment (kevinwallimann) [inline ui/src/app/components/workflows/workflow-form/dynamic-parts/string-sequence-part/string-sequence-part.component.ts:56]: Why is this needed now?

**Comment by kevinwallimann** [comment]
I like it in general. However, now we have a lot of dead code from the old implementation. Are you going to create a separate PR where you clean up all of the old code or what's your plan?

File: ui/src/app/components/workflows/workflow-form/jobs/jobs.component.ts (hunk 3 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/jobs.component.ts:25]: I would leave it this way for now. In the next PR I will get rid of the workflowFormParts, because I didn't want to do it in one PR

File: ui/src/app/services/workflow/workflow.service.ts (hunk 4 above)
Comment (jozefbakus) [reply ui/src/app/services/workflow/workflow.service.ts:175]: Yes, you are right like in previous comment, I would leave it and do it in the next PR

File: ui/src/app/components/workflows/workflow-form/dynamic-parts/string-sequence-part/string-sequence-part.component.ts (hunk 5 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/dynamic-parts/string-sequence-part/string-sequence-part.component.ts:56]: Because child components in job.component.html in ngSwitch are not destroyed when you switch between hyperdrive and hyperdrive form config.

File: ui/src/app/components/workflows/workflow-form/jobs/job/job.component.ts (hunk 2 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job.component.ts]: Done

File: ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts (hunk 1 above)
Comment (jozefbakus) [reply ui/src/app/components/workflows/workflow-form/jobs/job/job-types/hyperdrive/hyperdrive.component.ts]: Done

**Review by kevinwallimann — APPROVED** [review]
